shared/pub/metadata.js

```diff
@@ -0,0 +1,54 @@
+export const getPDFDownloads = (pub) => {
+	const downloads = pub.downloads;
+	const exports = pub.activeBranch.exports;
+	if (downloads && downloads.length > 0) {
+		return downloads.filter((download) => {
+			if (download.url.substring(download.url.length - 4) === '.pdf') {
+				return download;
+			}
+			return false;
+		});
+	}
+	if (exports && exports.length > 0) {
+		return exports.filter((exportFile) => {
+			if (exportFile.format === 'pdf') {
+				return exportFile;
+			}
+			return false;
+		});
+	}
+	return false;
+};
+
+export const getTextAbstract = (content) => {
+	let abstract = '';
+	if (!content) {
+		return abstract;
+	}
+	if (
+		content.content[0].type === 'heading' &&
+		content.content[0].attrs.level === 1 &&
+		content.content[0].attrs.id === 'abstract'
+	) {
+		content.content[1].content.forEach((item) => {
```

Comment:
⚠️ Likewise here. In general, there's no guarantee that a Prosemirror node will have a `content` member. (It took me months to remember all these rules and I still sometimes call it `children` by accident)